### PR TITLE
fix:enable default mouse/touch events on monaco content widget

### LINKED ISSUE #46166: Step content can't be interacted with using touch interface
### Describe the Issue

Recently we moved the Step instructions above the monaco editor using monaco's built in content widget (PR #45803) which solved an accessibility issue for screen reader users but has apparently created a new one for touch users. It looks like the monaco editor is intercepting touch events on the content widget and not allowing them to execute their default behaviors which is making it impossible for touch users to scroll code examples in the content or copy/paste from the content.

An example can be found on Step 18 - Cafe Menu. If you view this step using a touch device with a view port narrow enough to cause the code example to scroll you will not be able to scroll it with your finger. And you will not be able to copy any text in the content box.

Since I created the PR that moved the instructions into the content widget I of course feel obligated to fix this. That will teach me not to test on my phone!

UPDATE: Also, this issue is preventing the default right mouse click action from working on the content widget. Turns out monaco is not intercepting touch events but rather mouse events.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-basic-css-by-building-a-cafe-menu/step-18

### Your code

N/A

### Expected behavior

User should be able to interact with content using touch device.

### Screenshots

_No response_

### System

- Device: moto g6
- OS: android 9
- Browser: Chrome
- Version: 101

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bbsmooth**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes locally on my machine.

Closes #46166 

Unfortunately, I can find no way to fix this issue using the monaco API. I'm actually not surprised as they make it quite clear that the monaco editor is not supported in mobile browsers. So I had to hack this by adding some event listeners on the document to intercept certain mouse events before monaco could get to them, and then stop their propagation in order to allow their default behaviors to take place. This could be made a little more efficient by only adding the listeners for the new editor interface but I couldn't find a way to determine this. Please let me know if there is one.

Also, I have not been able to test this on an actual touch device yet. The monaco editor does not listen for any touch events as far as I can tell, so I believe just handling the mouse events will solve this problem. Is there a public facing dev server we can put these changes on so I can test from a phone?

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
> Is there a public facing dev server we can put these changes on so I can test from a phone?

GitPod is probably the easiest way to spin up a dev server. It's a bit of a clunky interface, but it's possible to open up the site in a new tab.

**Comment by bbsmooth** [comment]
Thanks for the suggestion @ojeytonwilliams. Never used gitpod before but it seems to work well. Unfortunately I also discovered that my fix didn't fix the touch screen issues, so more work to do.

**Comment by bbsmooth** [comment]
I was wrong, it appears that monaco is also capturing touch events, so I used this same trick of intercepting the touch events and now I am able to scroll the code examples and copy text in the content widget. But of course a new problem was created in doing this. Up/down scrolling no longer works properly if you swipe over the content widget. The best way to explain this is to give you an example. Using your phone, open [Cafe Menu Step 18](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-basic-css-by-building-a-cafe-menu/step-18). Swipe up/down over the instructions box and notice that the entire editor scrolls. Now scroll down to the Check Your Code button and click it to trigger the hint. Again, scroll up/down on the hint box and notice that the entire editor does NOT scroll. The difference here is that the hint box is in an "overlay widget" and monaco does not steal the touch events on it, but the instructions are in a "content widget" and monaco does steal the touch events on that. The ironic thing is that originally we had the instructions in an overlay widget and thus users were initially not able to swipe scroll over the instructions, so moving them to a content widget fixed that problem, but created this new problem.

I'm not sure what the solution is at this point. Could it have something to do with the way scrolling is being implemented on our end? Initially I thought a workaround for this would be to scroll on the left gutter where the line numbers are since the instructions box does not extend over that gutter. But apparently monaco is not stealing the touch events for that part either as scrolling over the gutter does not scroll the editor either. So I think there may be a bigger problem here since we also can't scroll the editor over the gutter or the hint overlay.

In summary, I have the fixes for the right mouse click and touch events on the content widget working but we need to figure out how to fix the scrolling issue in general because I think it is conceivable that someone could have the hint box taking up most of the view port and then it would be very hard for them to scroll back up to to make changes.

**Comment by bbsmooth** [comment]
Another thought I have here is that we could add a skinny invisible overlay widget on either side of the instructions content widget. Since the overlay widgets seem to allow touch users to scroll the editor this would create a narrow scroll area on each side of the instructions.

**Comment by bbsmooth** [comment]
This is now ready to be tested. I did add an invisible content widget over the line numbers so that the touch user is able to scroll the editor at any position swiping over the line numbers. I also had to override a few CSS properties on the editor in order to place it over the line numbers but I have not noticed any adverse effects.

This PR fixes the following issues.

For touch users:
- User is now able to select content in the description.
- User is now able to scroll code snippets in the description.
- User is now able to scroll editor at any position by swiping over line number gutter. User will continue to NOT be able to scroll editor if swiping over description or results/hint box. User was always able to scroll editor when swiping over any editable lines in editor and this should continue.

For mouse user:
- User in now able to right click on description and get browser's native menu.

These issues are beginning to come up in the forum so I would suggest we try to fast track this as much as possible. Cafe Menu Step 18 is a good one to test the touch issues on. I've been using gitpod to fire up a dev server I can reach from my phone. I have tested on both Chrome/Firefox on android. Though I don't expect any issues, it would be cool if someone could test with Safari on iOS.

**Review by raisedadead — APPROVED** [review]
A really good way to confirm would be to get this out the door. 

I did some quick sanity checks that non-touch devices are not affected (which is what we care about first, as a larger share of the audience).
